### PR TITLE
feat: get-app from folder in filesystem

### PR DESCRIPTION
**Pull request by gavindsouza**
Frappe apps can be installed from existing repositories in the filesystem

Screenshot 2020-02-05 at 1 05 19 PM

Closes https://github.com/frappe/bench/issues/777

### CODE REVIEW

File: bench/app.py [hunk 1]
```diff
@@ -1,55 +1,56 @@
-from __future__ import print_function
-import os
-from .utils import (exec_cmd, get_frappe, check_git_for_shallow_clone, build_assets,
-	restart_supervisor_processes, get_cmd_output, run_frappe_cmd, CommandFailedError,
-	restart_systemd_processes)
-from .config.common_site_config import get_config
-
-import logging
+import json, logging, os, re, shutil, subprocess, sys
```
Comment (Thunderbottom) [inline bench/app.py]: i dont think we should be going anti-pattern and doing multiple imports on single line.

File: bench/app.py [hunk 2]
```diff
@@ -82,47 +85,61 @@ def add_to_excluded_apps_txt(app, bench_path='.'):
...
+				cwd=os.path.join(bench_path, 'apps'))
+
+	return repo_name
+
+
+def get_app(url, branch=None, bench_path='.', skip_assets=False, verbose=False, postprocess=True):
+	"""Installs Frappe app from url (git repo or file system) in bench env"""
+
+	try:
+		if os.path.exists(url):
```
Comment (Thunderbottom) [inline bench/app.py]: instead of doing all this, can we add a flag `--local` to handle local files?

Comment (gavindsouza) [reply bench/app.py]: any specific reason for adding a `local` flag?

Comment (Thunderbottom) [reply bench/app.py]: its easier for the user to just add another flag than to add more complications by checking if the path exists locally before trying to clone from remote.

**Comment by gavindsouza** [comment]
Resolved in #900

**Pull request closed without merging**